Remove remaning roctracer references

dgaliffiAMD opened the pull request:
- `Findrocprofiler.cmake` and `Findroctracer.cmake` modules that are no longer used.
- Updated sample output in documentation that included references to roctracer.
- Updated Dockerfiles. No need to install the `rocprofiler-dev` and `roctracer-dev` packages.
- Removed references in source: comments, backends, internal list of shared libraries

dgaliffiAMD committed:
Remove unused cmake modules

- `Findrocprofiler.cmake` and `Findroctracer.cmake` modules that are no longer used.

wilephan-amd reviewed (changes requested):
I think there are more hooks that need to be removed from `cmake/Packages.cmake`

https://github.com/dgaliffiAMD/rocprofiler-systems/blob/1ab4184efb14e7091c577e74e49b315eceec33aa/cmake/Packages.cmake#L20-L23
```cmake
20 rocprofiler_systems_add_interface_library(rocprofiler-systems-roctracer
21                                           "Provides flags and libraries for roctracer")
22 rocprofiler_systems_add_interface_library(rocprofiler-systems-rocprofiler
23                                           "Provides flags and libraries for rocprofiler")
``` 

https://github.com/dgaliffiAMD/rocprofiler-systems/blob/1ab4184efb14e7091c577e74e49b315eceec33aa/cmake/Packages.cmake#L164-L171
```cmake
164     if(ROCPROFSYS_ROCM_VERSION_MAJOR GREATER_EQUAL 4 AND ROCPROFSYS_ROCM_VERSION_MINOR
165                                                          GREATER 3)
166         set(roctracer_kfdwrapper_LIBRARY)
167     endif()
168
169     if(NOT roctracer_kfdwrapper_LIBRARY)
170         set(roctracer_kfdwrapper_LIBRARY)
171     endif()
```

Edit: just ran `grep -rni roctracer` on the branch. Looks like this change isn't that trivial... Aside from the CMake stuff and the documentation, some of these also needs to be adjusted:

```
$ grep -rni roctracer | egrep -v 'Binary|^cmake\/|^docs\/'
docker/Dockerfile.rhel:31:        yum install -y rocm-dev rocm-smi-lib roctracer-dev rocprofiler-dev libpciaccess && \
docker/Dockerfile.ubuntu:42:        apt-get install -y rocm-dev rocm-smi-lib roctracer-dev rocprofiler-dev rccl-dev libpciaccess0 ${EXTRA_PACKAGES} && \
docker/Dockerfile.opensuse:38:        zypper --non-interactive install -y rocm-dev rocm-smi-lib roctracer-dev rocprofiler-dev rccl-devel libpciaccess0 && \
external/timemory/source/timemory/backends/roctx.hpp:28:#    include <roctracer/roctx.h>
external/timemory/source/timemory/components/hip/components.hpp:113:               "accurate than 'roctracer' for kernel timing";
scripts/rocprof-sys-leak.supp:10:leak:libroctracer64
source/bin/rocprof-sys-instrument/internal_libs.cpp:326:                                            "libroctracer64.so",
source/bin/rocprof-sys-instrument/rocprof-sys-instrument.cpp:363:                        "tbbmalloc_proxy|gotcha|libunwind|roctracer64|hsa-runtime|amdhip|"
source/bin/rocprof-sys-sample/impl.cpp:722:                                               "roctracer",
source/lib/core/argparse.cpp:547:                                  "rcclp", "rocm-smi",    "roctracer",  "rocprofiler",
source/lib/core/components/fwd.hpp:42:ROCPROFSYS_DECLARE_COMPONENT(roctracer)
source/lib/core/components/fwd.hpp:130:TIMEMORY_SET_COMPONENT_API(rocprofsys::component::roctracer, project::rocprofsys,
source/lib/core/components/fwd.hpp:163:TIMEMORY_METADATA_SPECIALIZATION(rocprofsys::component::roctracer, "roctracer",
source/lib/core/hip_runtime.hpp:37:// following must be included before <roctracer_hip.h> for ROCm 6.0+
source/lib/core/hip_runtime.hpp:46:#    include <roctracer/hip_ostream_ops.h>
source/lib/rocprof-sys/library/components/backtrace.cpp:129:        if(_lbl.find("roctracer_") != _npos) return -1;
source/lib/rocprof-sys/library/components/pthread_create_gotcha.cpp:124:        // update roctracer_data
source/lib/rocprof-sys/library/ptl.cpp:119:namespace roctracer
source/lib/rocprof-sys/library/ptl.cpp:130:}  // namespace roctracer
source/lib/rocprof-sys/library/ptl.cpp:143:    if(roctracer::get_thread_pool_state() == State::Active)
source/lib/rocprof-sys/library/ptl.cpp:145:        ROCPROFSYS_DEBUG_F("waiting for all roctracer tasks to complete...\n");
source/lib/rocprof-sys/library/ptl.cpp:147:            roctracer::get_task_group(i).join();
source/lib/rocprof-sys/library/ptl.cpp:151:        ROCPROFSYS_DEBUG_F("roctracer thread-pool is not active...\n");
source/lib/rocprof-sys/library/ptl.cpp:165:    if(roctracer::get_thread_pool_state() == State::Active)
source/lib/rocprof-sys/library/ptl.cpp:167:        ROCPROFSYS_DEBUG_F("Waiting on completion of roctracer tasks...\n");
source/lib/rocprof-sys/library/ptl.cpp:170:            roctracer::get_task_group(i).join();
source/lib/rocprof-sys/library/ptl.cpp:171:            roctracer::get_task_group(i).clear();
source/lib/rocprof-sys/library/ptl.cpp:172:            roctracer::get_task_group(i).set_pool(nullptr);
source/lib/rocprof-sys/library/ptl.cpp:174:        roctracer::get_thread_pool_state() = State::Finalized;
source/lib/rocprof-sys/library/ptl.cpp:178:        ROCPROFSYS_DEBUG_F("roctracer thread-pool is not active...\n");
source/lib/rocprof-sys/library/ptl.cpp:223:roctracer::get_task_group(int64_t _tid)
source/lib/rocprof-sys/library/ptl.cpp:228:    static thread_local auto& _v = (roctracer::get_thread_pool_state() = State::Active,
source/lib/rocprof-sys/library/ptl.hpp:61://      roctracer
source/lib/rocprof-sys/library/ptl.hpp:65:namespace roctracer
source/lib/rocprof-sys/library/ptl.hpp:69:}  // namespace roctracer
source/lib/rocprof-sys/library/thread_info.hpp:42://                backends, e.g., roctracer will create threads
tests/rocprof-sys-openmp-tests.cmake:48:        "${_ompt_environment};ROCPROFSYS_ROCTRACER_HSA_ACTIVITY=OFF;ROCPROFSYS_ROCTRACER_HSA_API=OFF"
```

I dunno how to look for the corresponding changes needed for `rocprofiler`, since `grep` wouldn't work as nicely...

dgaliffiAMD committed:
Update Dockerfiles

Removing roctracer-dev, rocprofiler-dev,and  rocm-smi-lib. They're no longer needed and also included in rocm-dev.

dgaliffiAMD committed:
Remove unused roctracer references from cmake

dgaliffiAMD committed:
Removing some more `roctracer` references from source.

dgaliffiAMD commented:
I still need to address the the changes to docs, but updating the commit get test coverage.

dgaliffiAMD committed:
Format source and cmake

dgaliffiAMD committed:
Update documentation

wilephan-amd commented:
Just noticed that explicit references to `roctx` also got removed, is that also now implemented as part of `rocprofiler-sdk`?

dgaliffiAMD commented:
> Just noticed that explicit references to `roctx` also got removed, is that also now implemented as part of `rocprofiler-sdk`?

Hi @wilephan-amd, that's correct. rocTX is now handled through librocprofiler-sdk-roctx. They are enabled through the ROCPROFSYS_ROCM_DOMAINS configuration setting adding "marker_api" or "roctx" (it's set by default). 

I understand your point. We should have a test case to explicitly test for them. Let me look into this. I can add it to a separate PR.

dgaliffiAMD commented:
Rebased with latest amd-staging

dgaliffiAMD committed:
Keep the old roctracer / rocprofiler libraries as internal
